test/cpp/jit/test_backend.cpp

```diff
@@ -32,9 +33,97 @@ TEST(BackendTest, ToBackend) {
   // lowered module
   auto lm = torch::jit::detail::codegen_backend_module(
       "test_backend", m, compile_spec, any_dict_ty);
+  // lowered module code:
+  /*
+class test_backendLoweredModule(Module):
+  __parameters__ = []
+  __buffers__ = []
+  __processed_module : Any
+  __method_compile_spec : Dict[str, Any]
+  __backend : __torch__.torch.classes.__backends__.test_backend
+  __handles : Dict[str, Any]
+  def __create_backend(self: torch.jit.test_backendLoweredModule) -> None:
+    _0 =
+__torch__.torch.classes.__backends__.test_backend.__new__(__torch__.torch.classes.__backends__.test_backend)
+    _1 = (_0).__init__()
+    self.__backend = _0
+    return None
+  def __getstate__(self: torch.jit.test_backendLoweredModule) -> Tuple[Dict[str,
+Any], Any]: _2 = (self.__method_compile_spec, self.__processed_module) return _2
+  def __setstate__(self: torch.jit.test_backendLoweredModule,
+    state: Tuple[Dict[str, Any], Any]) -> None:
+    self.__method_compile_spec = (state)[0]
+    self.__processed_module = (state)[1]
+    _3 = (self).__create_backend()
+    _4 = (self.__backend).compile(self.__processed_module,
+self.__method_compile_spec, ) self.__handles = _4 return None def forward(self:
+torch.jit.test_backendLoweredModule, x: Tensor, h: Tensor) -> Tuple[Tensor,
+Tensor]: _5 = uninitialized(Tensor) typed_inputs = annotate(List[Any], [x, h])
+    _6 = (self.__backend).execute((self.__handles)["forward"], typed_inputs, )
+    _7, _8, = _6
+    _9 = isinstance(_7, Tensor)
+    if _9:
+      _10 = unchecked_cast(Tensor, _7)
+    else:
+      ops.prim.RaiseException("AssertionError: ")
+      _10 = _5
+    _11 = isinstance(_8, Tensor)
+    if _11:
+      _12 = unchecked_cast(Tensor, _8)
+    else:
+      ops.prim.RaiseException("AssertionError: ")
+      _12 = _5
+    return (_10, _12)
+
+   */
   auto res = lm.forward(inputs).toTuple()->elements();
   AT_ASSERT(res[0].toTensor().equal(ref[0].toTensor()));
   AT_ASSERT(res[1].toTensor().equal(ref[1].toTensor()));
 }
+
+TEST(BackendTest, TestCompiler) {
```

Comment:
Just double checking, "TestCompiler" and "ToBackend" test the same. The only difference is that "TestCompiler" is simpler (single output vs tuple), correct?

Reply:
The major difference is their backends. ```ToBackend``` tests the "test_backend" backend, which does not have an actual compiler but everything is hard-coded. ```TestCompiler``` is testing a new backend, "backend_with_compiler_demo", to test the compilation of a "supported" feature.

Reply:
Got it. I missed the different backend class.
Maybe once you're done with the new dummy backend we don't need the other one.

Reply:
Good point. I need to have the python frontend test for this backend as well. After that is done we could remove the other one.